flang/lib/Lower/ConvertType.cpp

```diff
@@ -239,9 +239,9 @@ struct TypeBuilder {
     }
 
     if (isPtr || Fortran::semantics::IsPointer(symbol))
-      ty = fir::PointerType::get(ty);
+      ty = fir::BoxType::get(fir::PointerType::get(ty));
     else if (isAlloc || Fortran::semantics::IsAllocatable(symbol))
```

Comment:
The `isAlloc` value reflects that the symbol was promoted to the heap by the compiler, non? Effectively, it should be placed on the stack, but the compiler may have decided it was "big" and wants to move it from the stack frame to the heap to reduce stack resource demand... In that case, what was a true pointer would now have emboxing and unboxing overhead as well.

Reply:
Good point. `isAlloc` and `isHeap` are a bit blurry for me because I don't think they are used yet for non Fortran pointers/allocatables. Are they actually used for such promotion ?
Question about it though, will lowering be the one dealing with this stack to heap promotion or should a fir pass do that ?
I can imagine that fir may would have more visibility to make this choices after inlining and so on.

Reply:
Both cases. The bridge will automatically modify some variables storage properties (and should already be doing so). However, writing a pass also makes sense.